src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -711,6 +705,227 @@ public String[] executeSimple(Object[][] params) {
+			)
+			.examples(
+				"mean(1, 2, 3) = 2",
+				"mean(0, 5, 10) = 5",
+				"mean(13, 97, 376, 709) = 298.75"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("median", new Parameter[]{
+			new Parameter<>("numbers", DefaultClasses.NUMBER, false, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				AtomicBoolean invalid = new AtomicBoolean(false);
+				Number[] sorted = Arrays.stream(params[0])
+					.filter(object -> {
+						if (!(object instanceof Number number))
+							return false;
+						if (number instanceof Double doubleNumber && (doubleNumber.isInfinite() || doubleNumber.isNaN())) {
+							invalid.set(true);
+							return false;
+						}
+						return true;
+					})
+					.map(object -> (Number) object)
+					.sorted(((o1, o2) -> {
+						Double n1 = o1.doubleValue();
+						Double n2 = o2.doubleValue();
+						return n1.compareTo(n2);
+					}))
+					.toArray(Number[]::new);
+				if (invalid.get())
+					return null;
+				int size = sorted.length;
+				if (size % 2 == 1)
+					return new Number[]{sorted[Math2.ceil(size /2)]};
+				int half = size / 2;
+				double first = (sorted[half - 1]).doubleValue();
+				double second = (sorted[half]).doubleValue();
+				double median = (first+second)/2;
+				return new Number[]{median};
+			}
+		})
+			.description(
+				"Get the middle value of a sorted list of numbers. "
+				+ "If the list has an even number of values, the median is the average of the two middle numbers.",
+				"You cannot get the median of a set of numbers that includes infinity or NaN."
+			)
+			.examples(
+				"median(1, 2, 3, 4, 5) = 3",
+				"median(1, 2, 3, 4, 5, 6) = 3.5",
+				"median(0, 123, 456, 789) = 289.5"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<>("factorial", new Parameter[]{
+			new Parameter<>("number", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Double number = ((Number) params[0][0]).doubleValue();
+				boolean negative = false;
+				if (number < -170) {
+					return new Number[]{Double.NEGATIVE_INFINITY};
+				} else if (number < 0) {
+					negative = true;
+					number = Math.abs(number);
+				} else if (number <= 1) { // 0 and 1
+					return new Number[]{1};
+				} else if (number > 170) {
+					return new Number[]{Double.POSITIVE_INFINITY};
+				}
+				Double result = 1d;
+				for (double i = number; i > 1; i--) {
+					if (result.isInfinite() || result.isNaN())
+						break;
+					result *= i;
+				}
+				if (negative)
+					result *= -1d;
+				return new Number[]{result};
+			}
+		})
+			.description(
+				"Get the factorial of a number. ",
+				"Getting the factorial of any number above 20 goes into the Double range leading to floating points.",
```

Comment:
```suggestion
				"Getting the factorial of any number above 21 will return an approximation, not an exact value.",
```
21! happens to be precisely representable by doubles, but 22! is not. I did not check for FP error due to the multiplication though, you should double check that in the tests (focus on 21/20/19 etc)

Reply:
unresolved, tests still needed